presto-main/src/main/java/com/facebook/presto/operator/scalar/ReflectionParametricScalar.java

```diff
@@ -597,7 +654,6 @@ public MethodHandleAndConstructor specialize(Signature boundSignature, BoundVari
             if (nullable) {
                 return Primitives.wrap(clazz);
             }
-            checkArgument(clazz != void.class);
```

Comment:
What is the reasoning behind this check?
Why do we allow `Void.class` and don't allow `void.class`?
Why simple function for (I assume) wrapping type into it's nullable version if it is expected to be nullable is not agnostic about the type that it's taking? I think if we don't wan't `void` to be allowed, we should check that on higher level and for `Void` as well.